sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -325,6 +341,14 @@ trait Row extends Serializable {
   def getAs[T](i: Int): T = get(i).asInstanceOf[T]
 
   /**
+   * Returns the Optional value of a given fieldName.
+   * If the value is null or if it cannot be cast to the requested type, None is returned. None is
+   * also returned if either the schema is not defined or else the given fieldName does not exist
+   * in the schema.
+   */
+  def getAsOpt[T](fieldName: String): Option[T] = None
```

Comment:
I considered returning a `Try`. But there are a lot of cases, personally, where I don't care about the error since I know the kind of bad data I can expect. But that's just me.